### PR TITLE
Frostbite light falloff Step1

### PR DESCRIPTION
**Pull request by sebavan**
Please, do not merge, this requires more testing today :-)

But review if possible so that I integrate the changes in today.

This addresses the first step of #4148

### CODE REVIEW

File: src/Lights/babylon.spotLight.ts [hunk 1]
```diff
@@ -37,8 +43,29 @@
...
         public set angle(value: number) {
             this._angle = value;
+            this._cosHalfAngle = Math.cos(value * 0.5);
             this._projectionTextureProjectionLightDirty = true;
             this.forceProjectionMatrixCompute();
+            this._computeAngleValues();
+        }
+
+        /**
+         * Only used in Frostbite falloff mode, this defines the angle where
```
Comment (deltakosh) [inline src/Lights/babylon.spotLight.ts]: not sure we can use the Frosbite name as it is copyrighted

File: src/Materials/PBR/babylon.pbrBaseMaterial.ts [hunk 2]
```diff
@@ -164,6 +165,22 @@
      * http://doc.babylonjs.com/extensions/Physically_Based_Rendering
      */
     export abstract class PBRBaseMaterial extends PushMaterial {
+        /**
+         * PBRMaterialLightFalloff Physical: light is falling off folliwing the inverse squared distance law.
```
Comment (bghgary) [inline src/Materials/PBR/babylon.pbrBaseMaterial.ts]: spelling: following

**Comment by sebavan** [comment]
The new falloff can be tested on this playground (once deployed): http://playground.babylonjs.com/#20OAV9#195

**Comment by sebavan** [comment]
@bghgary You can now add the gltf loading part :-)

cc: @MiiBond

**Pull request merged**